docs/preloading-cloud-data.md

```diff
@@ -0,0 +1,100 @@
+* Instance type names
+* Common instance type details (vCPUs, cores, memory, local drive)
+* Specific type details (VM generation for Azure)
+* Supported flag (when type meets Minimum RHEL Requirements criteria)
+
+The data is stored as [YAML](./../internal/preload) and available through a `preload` Go package and REST API endpoints.
+
+## Model
+
+The terminology is different for each hyperscaler, however, the data model uses the AWS terminology: regions and (availability) zones:
+
+* AWS EC2: zones (`us-east-1`) and availability zones (`a`), fully qualified name is `us-east-1a`
+* Azure: locations (`northeurope`) and availability zones (`1`), fully qualified name is `northeurope_1`
+* GCP: zones (`us-west`) and regions (`b`), fully qualified name is `us-west-b`
+
+### Registered Instance Types
+
+The `InstanceType` type contains common fields like number of vCPUs and cores, total amount of memory, ephemeral storage and supported flag. It also has an architecture, therefore, a single instance type cannot have multiple architectures. In this model, every instance type with a different architecture has a different instance type name. This applies to all major hyperscalers except Azure where Intel 64bit and 32bit are shared. However, 32bit is no longer supported by RHEL and therefore this is not an issue.
+
+The only supported architectures at the moment are: x86_64 and arm64. All other instance types are ignored. InstanceType also contains optional details, currently for Azure the VM generation. 
+
+Functions in the `preload` package can be used to find particular `InstanceType` by name.
+
+### Regional Type Availability
+
+Depending on hyperscaler provider, different instance types can be available in different regions. For this reason, a function in the `preload` package will return a slice of instance types for every zone and region pair. The data is stored in individual files as [YAML](./../internal/preload) with the following naming convention:
+
+* AWS EC2: `region.yaml` (all availability zones have the same instance types)
+* Azure: `location_zone.yaml` (different instance types in zones)
+* GCP: `zone.yaml` (all)
+
+## Refreshing data
+
+In order to update the embedded data with latest available instance types, use provided `typesctl` utility and Makefile targets. First, configure connections:
+
+```
+# cat config/api.env
+AWS_KEY=KEY
+AWS_SECRET=SECRET
+AZURE_TENANT_ID=UUID
+AZURE_SUBSCRIPTION_ID=UUID
+AZURE_CLIENT_ID=UUID
+AZURE_CLIENT_SECRET=SECRET
+GCP_JSON=xxxxxxx
+GCP_PROJECT_ID=myproject-12345
+```
+
+Important: AWS requires special attention: global endpoint session setting must be set for an account. 
+
+1. Sign in as a root user or a user with permissions to perform IAM administration tasks. To change the compatibility of session tokens, you must have a policy that allows the iam:SetSecurityTokenServicePreferences action.
+2. Open the IAM console. In the navigation pane, choose Account settings.
```

Comment:
typo

Reply:
Where? (I copy pasted this from AWS docs.)

Reply:
pane ?